File: Stack/1544. Make The String Great/1544.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack=[]
        for i in range(len(s)):
            if stack and stack[-1].upper()==s[i].upper() and ord(stack[-1])-ord(s[i])!=0:
                stack.pop()
            else:
                stack.append(s[i])
        return "".join(stack)

File: Stack/1544. Make The String Great/1544.cpp
#pragma GCC optimize('0fast', 'unroll-loops')

class Solution {
public:
    std::string makeGood(std::string s) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        std::stack<char> st;
        for(char c : s) {
            if(!st.empty() && (std::toupper(st.top()) == std::toupper(c)) && (st.top() != c)) {
                st.pop();
            } else {
                st.push(c);
            }
        }

        std::string res = "";
        while(!st.empty()) {
            res = st.top() + res;
            st.pop();
        }
        return res;
    }
};
